### PR TITLE
Maintainer sample transfer project listing fix

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR fixes the maintainer sample transfer which would list all projects a user had manageable access to instead of just listing the projects in the hierarchy of the current project. The fix was to pass the correct parameter (project namespace) to the user_has_maintainer_access? method in the member model

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com`
2. Create a group `Group A` with 2 subgroups
3. Add a project to `Group A` and to each of it's subgroups
4. Add a sample to each project
5. Add the `user1@email.com` as a `Maintainer` to `Group A` -> `Subgroup 1` -> `Project 2`
6. Sign out and login as the `user1@email.com`
12. Go into  `Group A` -> `Subgroup 1` -> `Project 2`
13. Go to the samples page for the project
14. Select a sample and click `Transfer`
15. In the Transfer Samples modal within the dropdown you should no projects for the user to transfer into as no other projects exist in the hierarchy
16. Add another project to `Group A` -> `Subgroup 1` 
17. Repeat step 12 but for the new project created in step 16, through step 15. You should see the project from step 16 in the dropdown
18. Sign out and login as `admin@email.com`
19. Create a group `Group B` and a project within and a sample within the project
20. Add `user1@email.com` as a `Maintainer` to the project in step 19
21. Sign out and login as `user1@email.com`
22. Go into the project from step 19
23. Go to the samples page, select a sample, and click `Transfer`
24.  In the Transfer Samples modal within the dropdown you should no projects for the user to transfer into as no other projects exist in the hierarchy

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by ericenns** [comment]
This is working as expected now, but when I have an empty dropdown for project selection and click submit I get the following error message which needs updating. Would you like to fix it in this PR or a different one, see image below.

image

**Comment by deepsidhu85** [comment]
> This is working as expected now, but when I have an empty dropdown for project selection and click submit I get the following error message which needs updating. Would you like to fix it in this PR or a different one, see image below.
> 
> image

Updated in 2d5c740b84e01974c2c5e838ba83775445389491

image

File: app/views/projects/samples/transfers/_transfer_modal.html.erb [hunk 1]
```diff
@@ -19,6 +19,7 @@
         <div class="flex flex-row-reverse">
           <%= form.submit t("projects.samples.transfers._transfer_modal.submit_button"),
                       class: "button button--size-default button--state-primary",
+                      disabled: !@projects.count.positive?,
```
Comment (ericenns) [inline app/views/projects/samples/transfers/_transfer_modal.html.erb:22]: **nitpick (blocking):** Can you change this to `@projects.count.zero?`

**Review by ericenns — CHANGES_REQUESTED** [review]
Just one minor comment otherwise this is good

File: app/views/projects/samples/transfers/_transfer_modal.html.erb (hunk 1 above)
Comment (deepsidhu85) [reply app/views/projects/samples/transfers/_transfer_modal.html.erb]: Sure thing. updated in 9fa93bc4280c6f9ae701c65f36d8b5d76e577485

**Review by ericenns — APPROVED** [review]

**Pull request merged**